pysrc/bytewax/_encoder.py

```diff
@@ -0,0 +1,53 @@
+import datetime
+import json
+import types
+
+from .bytewax import Dataflow
+from .inputs import ManualInputConfig, InputConfig
+from .window import ClockConfig, WindowConfig, SystemClockConfig
+from .outputs import (
+    ManualOutputConfig,
+    ManualEpochOutputConfig,
+    StdOutputConfig,
+    KafkaOutputConfig,
+)
+
+
+class DataflowEncoder(json.JSONEncoder):
+    """Custom JSON encoder for a Dataflow
+
+    This class is used in conjunction with the Rust `webserver` module to
+    produce a JSON representation of a bytewax Dataflow.
+
+    __getstate__() is a method defined on all of the Python classes we
+    create in Rust to return a PyDict representation of that class.
+    """
+
+    def default(self, obj):
+        if "__getstate__" in dir(obj):
```

Comment:
I think a slightly more robust way to do this is [`hasattr`](https://docs.python.org/3/library/functions.html#hasattr) (and the way I've seen in the past), given all the hesitation we've had surrounding `dir`.